examples/mk_simple_certs.py

```diff
@@ -7,15 +7,18 @@
 cakey = createKeyPair(TYPE_RSA, 2048)
 careq = createCertRequest(cakey, CN='Certificate Authority')
 cacert = createCertificate(careq, (careq, cakey), 0, (0, 60*60*24*365*5)) # five years
+
 print('Creating Certificate Authority private key in "simple/CA.pkey"')
-open('simple/CA.pkey', 'w').write(crypto.dump_privatekey(crypto.FILETYPE_PEM, cakey))
+open('simple/CA.pkey', 'w').write(crypto.dump_privatekey(crypto.FILETYPE_PEM, cakey).decode('utf-8'))
```

Comment:
He meant a context manager:

```py
with open(...) as f:
    f.write(...)
```

On Wed, Apr 29, 2015 at 10:25 PM, elitest <notifications@github.com> wrote:

> In examples/mk_simple_certs.py
> <https://github.com/pyca/pyopenssl/pull/245#discussion_r29399782>:
>
> >  print('Creating Certificate Authority private key in "simple/CA.pkey"')
> > -open('simple/CA.pkey', 'w').write(crypto.dump_privatekey(crypto.FILETYPE_PEM, cakey))
> > +open('simple/CA.pkey', 'w').write(crypto.dump_privatekey(crypto.FILETYPE_PEM, cakey).decode('utf-8'))
>
> Thanks for the feedback @hynek <https://github.com/hynek>. I thought
> contexts were for creating connections with certs that had already been
> created. Or am I missing something?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/pyopenssl/pull/245/files#r29399782>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084